### ISSUE TITLE
Alternative OSD/OSK for Wayland?

### ISSUE THREAD

**Issue by andersmmg** [kozec/sc-controller#689]
I know there's issues with creating an overlay on Wayland, but would it be possible to use an actual window? Like have an option when running Wayland to have a window pop up over whatever you're doing to change controller modes etc. Not sure if that would work for the keyboard... but it would be a start.

**Comment by kozec** [comment]
Not really, there's no way to tell wayland compositor to create always on top window.  From my understanding, most of features needed for OSD/OSK are intentionally _not_ part of the protocol.

**Comment by StaticRocket** [comment]
@kozec this has changed with the addition of the following extensions:

- layer-shell
- virtual-keyboard-v1

[Squeekboard](https://gitlab.gnome.org/World/Phosh/squeekboard) has a working example but it does require a little compositor interaction at the moment.
